sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -783,7 +783,7 @@ class SQLContext(@transient val sparkContext: SparkContext)
 
   /** Returns the specified table as a [[DataFrame]]. */
   def table(tableName: String): DataFrame =
-    DataFrame(this, catalog.lookupRelation(Seq(tableName)))
+    DataFrame(this, UnresolvedRelation(Seq(tableName)))
```

Comment:
I don't see why.  Eager analysis is only for debugging complex failures in the analyzer.  Missing a table doesn't really fall into that category.